plugins/modules/ndo_service_device_cluster.py

```diff
@@ -0,0 +1,641 @@
+        description:
+        - IPSLA monitoring policy configuration.
+        type: dict
+        suboptions:
+          name:
+            description:
+            - Name of the IPSLA monitoring policy.
+            type: str
+          template:
+            description:
+            - Template for the IPSLA monitoring policy.
+            - This parameter or O(interface_properties.ipsla_monitoring_policy.template_id) is required.
+            type: str
+          template_id:
+            description:
+            - ID of the template for the IPSLA monitoring policy.
+            - This parameter or O(interface_properties.ipsla_monitoring_policy.template) is required.
+            type: str
+      ipsla_monitoring_policy_uuid:
+        description:
+        - UUID of the IP SLA monitoring policy.
+        type: str
+      qos_policy:
+        description:
+        - Quality of Service (QoS) policy configuration.
+        type: dict
+        suboptions:
+          name:
+            description:
+            - Name of the QoS policy.
+            type: str
+          template:
+            description:
+            - Template for the QoS policy.
+            - This parameter or O(interface_properties.qos_policy.template_id) is required.
+            type: str
+          template_id:
+            description:
+            - ID of the template for the QoS policy.
+            - This parameter or O(interface_properties.qos_policy) is required.
+            type: str
+      qos_policy_uuid:
+        description:
+        - UUID of the QoS policy.
+        type: str
+      preferred_group:
+        description:
+        - Whether the interface belongs to a preferred group.
+        type: bool
+      rewrite_source_mac:
+        description:
+        - Whether to rewrite the source MAC address.
+        type: bool
+      anycast:
+        description:
+        - Indicates if anycast is enabled.
+        type: bool
+      config_static_mac:
+        description:
+        - Indicates if static MAC configuration is enabled.
+        type: bool
+      is_backup_redirect_ip:
+        description:
+        - Indicates if it is a backup redirect IP.
+        type: bool
+      load_balance_hashing:
+        description:
+        - Load balancing hashing method.
+        type: str
+        choices: ["sourceDestinationAndProtocol", "sourceIP", "destinationIP"]
+      pod_aware_redirection:
+        description:
+        - Indicates if pod-aware redirection is enabled.
+        type: bool
+      resilient_hashing:
+        description:
+        - Indicates if resilient hashing is enabled.
+        type: bool
+      tag_based_sorting:
+        description:
+        - Indicates if tag-based sorting is enabled.
+        type: bool
+      min_threshold:
+        description:
+        - Minimum threshold value for redirect.
+        type: int
+      max_threshold:
+        description:
+        - Maximum threshold value for redirect.
+        type: int
```

Comment:
for threshold values we are currently providing valid values in the documentation, should we add this here?

Reply:
ok